### PR TITLE
Fix export default regexs

### LINKED ISSUE #1798: Issues with default anonymous exports in 0.52.2
Some issues look to be caused by the regular expression [here](https://github.com/rollup/rollup/pull/1787/files#diff-6db9cc4c7dda84e33ed4bb1d66cb4587R49) being too narrow. It's possible to have anonymous default function/class definitions that do not match that pattern, and so `.match` returns null, and subscripting that throws an exception.

This can be reproduced with a default anonymous export of an async function, or by inserting a comment between two of the keywords in the definition.

Async functions would probably be able to be handled by extending the regular expression - but that doesn't sound like a good idea for dealing with comments. My feeling is that it would be better to be looking at the AST here rather than the string, but I haven't looked into this much.

### PR DESCRIPTION
**Pull request by nicolo-ribaudo**
Fixes #1798

I updated the regular expressions to match comments, async functions and generators.
They are built at runtime to make them more readable; they are built only once so the performance overhead shouldn't be noticeable.

The built regexes are these:
```js
/^(?:\s|\/\/.*[\n\r]|\/\*[^]*?\*\/)*export(?:\s|\/\/.*[\n\r]|\/\*[^]*?\*\/)+default(?:\s|\/\/.*[\n\r]|\/\*[^]*?\*\/)*/
/^(?:\s|\/\/.*[\n\r]|\/\*[^]*?\*\/)*export(?:\s|\/\/.*[\n\r]|\/\*[^]*?\*\/)+default(?:\s|\/\/.*[\n\r]|\/\*[^]*?\*\/)+(?:(?:async(?:\s|\/\/.*[\n\r]|\/\*[^]*?\*\/)+)?function(?:(?:\s|\/\/.*[\n\r]|\/\*[^]*?\*\/)*\*)?|class)/
```

### CODE REVIEW

File: test/form/samples/export-default-anonymous-declarations/async.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+// aaaa
+
+/* foo */
+
+default // jsjjjsjjjjsjs
+
+/* jsjjddjksj
+*/ // Too many comments lol
+
+async /* [no LineTerminator here] */ function
```
Comment (kzc) [inline test/form/samples/export-default-anonymous-declarations/async.js]: `acorn@5.2.1` parse of this source file yields:
```
Unexpected token (13:37)
```

Comment (kzc) [reply test/form/samples/export-default-anonymous-declarations/async.js]: Nevermind - I forgot the `--ecma2018` flag.

**Comment by adrianheine** [comment]
Note that Acorn doesn't parse async iterators yet (since they are stage3), but I wrote [a plugin](https://github.com/adrianheine/acorn-async-iteration) for it.

**Comment by nicolo-ribaudo** [comment]
I didn't write a test for async generators, but that regex should work for them.

**Pull request merged**